rust/ql/src/queries/security/CWE-327/BrokenCryptoAlgorithm.qhelp

```diff
@@ -3,20 +3,37 @@
 "qhelp.dtd">
 <qhelp>
      <overview>
-          <p>
-               Using broken or weak cryptographic algorithms can leave data
-               vulnerable to being decrypted or forged by an attacker.
-          </p>
+		<p>
+			Using broken or weak cryptographic algorithms may compromise
+			security guarantees such as confidentiality, integrity, and
+			authenticity.
+		</p>
 
-          <p>
-               Many cryptographic algorithms provided by cryptography
-               libraries are known to be weak, or flawed. Using such an
-               algorithm means that encrypted or hashed data is less
-               secure than it appears to be.
-          </p>
+		<p>
+			Many cryptographic algorithms are known to be weak or flawed. The
+			security guarantees of a system often rely on the underlying
+			cryptography, so using a weak algorithm can have severe consequences.
+			For example:
+		</p>
+
+		<ul>
+			<li>
+			If a weak encryption algorithm is used, an attacker may be able to
+			decrypt sensitive data.
+			</li>
+			<li>
+			If a weak hashing algorithm is used to protect data integrity, an
+			attacker may be able to craft a malicious input that has the same
+			hash as a benign one.
+			</li>
```

Comment:
It feels a bit odd that we call out weak hashing algorithms for Rust, when this query explicitly does not cover weak hashing and we have another Rust query (`rust/weak-sensitive-data-hashing`) which does - and which has its own detailed `.qhelp`.

That said, I could imagine digital signature algorithms (mentioned below) being bolted onto this query in future.  And I see the value in having similar text across different language `.qhelp`s.

Reply:
I think you're right that it's confusing. I've removed it from those languages where this query doesn't cover hashing. The qhelp files are still substantially the same. We can work towards having a separate hashing query for all languages.

I actually didn't find the weak sensitive data hashing query for rust because it's in a different folder. For all the other languages it's in the same folder. I think using CWE-328 for it makes sense. It's a shame that it's inconsistent with all the other languages. Another thing that we ought to harmonise one day.

Reply:
Here: https://github.com/github/codeql/pull/20649